### PR TITLE
feat(traffic_creator): create traffic and allocate locations

### PR DESCRIPTION
**Pull request by atqaamir**

### CODE REVIEW

File: simulation/vehicle/traffic_creator.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+from simulation.vehicle.vehicle import Vehicle
+import random
+
+
+class TrafficCreator(object):
+    __percentage_aggressive = ConfigReader.get_data("driving.traffic.driver_profile_type.aggressive")
+    __percentage_moderate = ConfigReader.get_data("driving.traffic.driver_profile_type.moderate")
+    __percentage_defensive = ConfigReader.get_data("driving.traffic.driver_profile_type.defensive")
+    __traffic = []
+    __instance = ""
```
Comment (internaccounts123) [inline simulation/vehicle/traffic_creator.py]: Initialize it with None

File: simulation/vehicle/traffic_creator.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            __instance = TrafficCreator(map1)
+        return __instance
+
+    @staticmethod
+    def __create_traffic(map1):
+
+        taken = []
+        for i in range(ConfigReader.get_data("driving.traffic.traffic_amount") * TrafficCreator.__percentage_aggressive):
+
+            v = Vehicle(ConfigReader.get_data("driving.aggressive.perception_size"),
```
Comment (Nabeel07925) [inline simulation/vehicle/traffic_creator.py]: Please refactor this.

File: simulation/vehicle/traffic_creator.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    def __init__(self, map1):
+
+        if TrafficCreator.__instance is None:
+            TrafficCreator.__traffic = TrafficCreator.__create_traffic(map1)
+        else:
+            raise Exception("Object already exists")
+
+    @staticmethod
+    def get_instance(map1):
```
Comment (Nabeel07925) [inline simulation/vehicle/traffic_creator.py]: Is this implementation related to some decision in the workflow? I mean to create an instance of the traffic creator.

File: common/utility.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+M_PI = 3.141519
```
Comment (Nabeel07925) [inline common/utility.py]: You can use numpy.pi to get the value of pi. That's more appropriate

**Review by Nabeel07925 — APPROVED** [review]
